### ISSUE TITLE
ReactiveCocoa2.5  build faild in Xcode Version 7.3 (7D175)

### ISSUE THREAD

**Issue by Sdoy** [ReactiveCocoa/ReactiveCocoa#2761]
![3f344c2f-8dbe-4597-b07a-548a0b72c42b](https://cloud.githubusercontent.com/assets/10302939/13940970/93497b58-f01d-11e5-9211-a96927537365.png)

**Comment by iworkcloud** [comment]
I am also seeking solutions

**Comment by iworkcloud** [comment]
Do not put the pre-compiled headers can be

**Comment by lacklock** [comment]
@iworkcloud how?

**Comment by lacklock** [comment]
after  upgraded to 7.3  i have the same problem .

**Comment by shenhualxt** [comment]
I am also seeking solutions

**Comment by lacklock** [comment]
https://forums.developer.apple.com/thread/38934

**Comment by jjksam** [comment]
Can I just replace all the __weak NSObject * to id to solve this ? Not sure about that, but that can make the problem pass the compilation.

**Comment by lacklock** [comment]
@jjksam yes,it's worked,juet delete __weak.

**Comment by undefined-zola** [comment]
http://stackoverflow.com/questions/4728225/is-it-ok-to-remove-prefix-pch-file-from-the-xcode-project

remove pre-compiled headers

**Comment by lacklock** [comment]
@undefined-zola ths , i get it

**Comment by NagiYan** [comment]
edit like this
#if __has_feature(objc_arc)
- (RACSignal *)rac_valuesForKeyPath:(NSString *)keyPath observer:(__weak NSObject *)observer;
#else
- (RACSignal *)rac_valuesForKeyPath:(NSString *)keyPath observer:(NSObject *)observer;
#endif

**Comment by kernel-lj** [comment]
delete the cocoapods location .h file which have __weak, it is a temporary  solutions，the best way is waiting 3rd party libraries update!

**Comment by Sdoy** [comment]
agree @kernel-lj ,the best way is waiting 3rd party libraries update,  because  not sure if it will cause memory leaks after  replace all the __weak NSObject * or use other ways to stop the Xcode complain

**Comment by willonboy** [comment]
@NagiYan thank you !

**Comment by WithdrwnKing** [comment]
why i can't unlock NSObject+RACPropertySubscribing.h.

**Comment by gcox** [comment]
You can also avoid this by not importing ReactiveCocoa in your precompiled header.

**Comment by gcox** [comment]
If removing from your pch file is too painful, you can do this:

```
#if __has_feature(objc_arc)
#import <ReactiveCocoa/ReactiveCocoa.h>
#endif
```

**Comment by Xummer** [comment]
@gcox GoodJob!

**Comment by wang9262** [comment]
@gcox For me,the error still appers.

**Comment by gcox** [comment]
@wang9262 You may be importing ReactiveCocoa in another header that you're importing in your pch file...you could try wrapping all imports in your pch file with the ```#if __has_feature(objc_arc)``` check.

**Comment by Vanson** [comment]
Please set 'Weak References in Manual Retain Release:YES'.

**Comment by wang9262** [comment]
@gcox Thanks.
 After we've done this 
```
#if __has_feature(objc_arc)
#import <ReactiveCocoa/ReactiveCocoa.h>
#endif
```
Some files that use RAC throw some the error like "Expected a type".At last we use `@class RACxxx` which solved this problem. Guess that there some files are marked as `-fno-objc-arc` in our project which caused this problem.

**Comment by jjksam** [comment]
@Vanson This work around is simple and efficient, thanks!

**Comment by mdiep** [comment]
Closing this because there doesn't seem like there's any work to be done in RAC.

tl;dr: In Xcode 7.3, `__weak` causes errors in files compiled as `-fno-objc-arc`. Since RAC uses `__weak`, you cannot use it in those files without setting the `Weak References in Manual Retain Release` setting to `YES`. If you're using a `.pch` that imports RAC, you're more likely to see this error.
